v1 react16 compat

bionikspoon opened the pull request:
Fixes #1715, Fixes #1703

- remove references to `React.PropTypes`
- fixed other removed imports

Note: Meant to be based on `tag:1.3.4`

braxtonchristensen commented:
Is there any plans for this to be released soon my team and I are really excited about React 16 and we love react-toolbox, thanks for all the work you guys put into it. It truly is a great component library.

bionikspoon commented:
@rubenmoya I added a commit to this, removed the dependency on my forked version of `react-css-themr`.  Should be merge friendly now. 

The applicable breaking change with `react-css-themr@2.x` was the removal of the `withRef` option and `getWrappedInstance` function.  In the new api, you can access the wrapped ref with `innerRef` callback prop.

lud2k commented:
Any chance this will be merged soon?

theskillwithin commented:
+1

rubenmoya commented:
Hi! Sorry for the delay. It looks good, although we'll have to take a closer look at the `react-themr` change since as you said, it's a breaking change and we have to know the impact it could have, but my plan is to have this merged ASAP.

theskillwithin commented:
Is this going to happen anytime soon?

mlcohen commented:
@rubenmoya Any input on the progress of this would be greatly appreciated. We currently use react-tool v1.3.4 and we'd like to upgrade our projects to use react v16.

graydonpleasants commented:
Can we get this merged?

mlcohen commented:
@graydonpleasants See https://github.com/react-toolbox/react-toolbox/pull/1814#issuecomment-369008874

mgcrea commented:
@rubenmoya you could release it as a v1.4.0-beta release instead of a patch if there is some uncertainty. Will gladly help you test such release. We've been waiting for this for so long...!!

MrEfrem commented:
hm, project stagnate...

theskillwithin commented:
ya its getting pretty slow.  they need to add more collaborators maybe.

mlcohen commented:
@theskillwithin Agree, however, one major blocker is that (currently) only @javivelasco is able to publish new packages to npm 😕

theskillwithin commented:
sucks cuz project owners work so hard and it sucks to get complaints from people benefiting from free stuff. but this is getting really bad,  I can't in good faith use this awesome library anymore.

mitoyarzun commented:
@rubenmoya , now that `2.0.0-beta.13` is out, any chance to have this one published? Thanks!

frankforpresident commented:
That would be nice indeed

rubenmoya commented:
I could release this as `1.3.5` although it scares me a little bit since there are changes to `react-themr` which I don't know if they are going to break something, keep in mind that the `v1` is not officially maintained anymore because of the `v2`.

Do you want me to release it anyway?

mgcrea commented:
@rubenmoya yes please 👍 !

Not having a React16 support is really a major burden for big apps that were built with toolbox v1. Many apps can't be rewritten for V2 as it would be a tremedous amount of work.

Ideally a beta `v1.3.5-beta.0` would be perfect. It will enable people to easily test against this new version with their specific workflow. If something breaks with `react-themr` we could always backport fixes to a fork, I will personnaly try to make it work and willing to commit time and eventual PRs to have a v1.x branch working with React16.

Thank you!

mitoyarzun commented:
I forked `react-css-themr@1.7.2` and `react-toolbox@1.3.4` under different names:

https://www.npmjs.com/package/react-css-themr-legacy
https://www.npmjs.com/package/react-toolbox-legacy

Repos are https://github.com/mitoyarzun/react-css-themr and https://github.com/mitoyarzun/react-toolbox.

Both are compatible with `react@16.4.1` (I haven't tested with other versions), and the docs work fine. I'm in the process of upgrading our apps to use those packages, so far I've found no difficulties (other than replacing the package names in the import statements.

Feel free to submit PRs if you find any bugs, my intention is to have give us the chance to upgrade to React 16 while we (now more calmly) move to an different solution.

mgcrea commented:
@mitoyarzun Awesome work! I've started to migrate a couple of projects and one thing I noticed was that the `Tabs` component does not seem to be working anymore, have you encountered something like this?

mitoyarzun commented:
@mgcrea please open an issue here: https://github.com/mitoyarzun/react-toolbox/issues, we don't use Tabs but I'll take a quick look.

manuphatak commented:
If I need to do anything let me know.
